### PR TITLE
Add with-lsf-libdir switch to OpenMPI

### PR DESCRIPTION
**Pull request by robertsawko**
As outlined in #44255, I've been struggling with getting OpenMPI compile with LSF support on a RHEL systems. Adding `--with-lsf-libdir` seems to solve my issue.

### CODE REVIEW

**Comment by robertsawko** [comment]
@spackbot help

**Review by hppritcha — APPROVED** [review]

**Comment by robertsawko** [comment]
I just clicked "Ready for review", but I confess I don't understand how `libs.directories[0]` picks up the right path. I have confirmed by running `spack install --keep-stage` and `ompi_info | grep ^Configure` that it does indeed pick up the right path, but I just don't understand how it happens.

I noticed that CUDA is being configured in a similar way.

**Review by alalazo — APPROVED** [review]

**Comment by robertsawko** [comment]
@alalazo can you explain to me what happens now? You assigned the PR to yourself? Do you expect me to keep this up to date with the current branch as we wait for reviews from all maintainers?

**Comment by alalazo** [comment]
@robertsawko No. This was set on auto-merge, but a required test is failing. It seems `py-torch` is taking more than 6 hrs to build in one of our pipelines?!? 

I'll try to re-run the pipelines (you could have done the same, by asking spackbot as I'll do in the comment below). If the error persists we need to investigate the failure.

**Comment by alalazo** [comment]
@spackbot run pipeline

**Pull request merged**
